### ISSUE TITLE
Add automated test for checking license notice on every source file

### ISSUE THREAD

**Issue by santisoler** [fatiando/community#33]
**Description**

Since fatiando/maintenance#10 we want that every source file (`.py` file) to have a license and copyright notice at the top.
Although adding the notice to every file is very straight forward through some scripting, I can definitely see myself forgetting to add the notice to any new file I add to any library.

Would be nice to have an automated test on CIs that checks if every `.py` file has the copyright notice.
Also would be very nice to add a new task on the `format` target: to run the script that adds the license notice to every `.py` file that doesn't have it.

**Apply changes to**

Libraries:

- [x] [boule](https://github.com/fatiando/boule)
- [x] [harmonica](https://github.com/fatiando/harmonica)
- [x] [pooch](https://github.com/fatiando/pooch)
- [x] [verde](https://github.com/fatiando/verde)

**Further instructions**

In order to solve this issue we need to start opening Pull Requests on each
repository listed above. Optionally, we can open Issues on each repository if
further discussion specific to that repository is needed.

Remember to mention this Issue on every Issue or Pull Request opened on each
repository for keeping a record by adding something like:

```
Related to fatiando/maintenance#12
```

A quick checklist:

1. Open an Issue on each repository listed above (optional). Mention this Issue
   on them.
2. Open a Pull Request on the same repository to tackle the Issue. Mention this
   Issue on them.
3. Check-off the repository on the list above once the Pull Request is merged.
4. Close this issue when all items are checked-off.

**We want your help**

We know that maintenance tasks are very demanding, so we don't expect a single
person to tackle this issue by themselves. Any help is very welcomed, so please
comment below that you want to take care of the changes on any repository and
we will assign it to you.

**Comment by leouieda** [comment]
This is pretty much done across the board. Closing and we can open a new one if we decide to go with a plugin or something else.
